### PR TITLE
Buildspec and GitHub status for GitHub bot

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/parse_partner_developers.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def main():
 
     if partner_dev:
         LOGGER.info(f"PARTNER_DEVELOPER: {partner_dev.upper()}")
-        LOGGER.info(f"PR_NUMBER: {os.getenv('PR_NUMBER', os.getenv('CODEBUILD_SOURCE_VERSION', '')).replace('/', '-')}")
+        LOGGER.info(f"PR_NUMBER: {os.getenv('PR_NUMBER', '').replace('/', '-')}")
```
Comment (arjkesh) [inline src/parse_partner_developers.py:34]: probably want to add "pr-" here

File: src/utils.py [hunk 2]
```diff
@@ -380,10 +380,8 @@ def build_setup(framework, device_types=None, image_types=None, py_versions=None
     enable_build = is_build_enabled()
 
     if build_context == "PR":
-        pr_number = os.getenv("CODEBUILD_SOURCE_VERSION")
+        pr_number = os.getenv("PR_NUMBER")
```
Comment (arjkesh) [inline src/utils.py:383]: you might need to make this an int

File: src/send_status.py [hunk 3]
```diff
@@ -62,10 +61,17 @@ def post_status(state):
...
 
     project_name = utils.get_codebuild_project_name()
-    trigger_job = os.getenv("TEST_TRIGGER", "UNKNOWN-TEST-TRIGGER")
     target_url = get_target_url(project_name)
-    context = f"{trigger_job}_{project_name}"
-    description = set_build_description(state, project_name, trigger_job)
+
+    test_context = os.getenv("TEST_TYPE")
+
+    if test_context and test_context not in "quick_checks":
```
Comment (junpuf) [inline src/send_status.py:68]: let's use `==` if you mean to check they are equal.

**Review by junpuf — APPROVED** [review]
